clients/fides-js/src/lib/preferences.ts

```diff
@@ -96,18 +96,21 @@ export const updateConsentPreferences = async ({
   Object.assign(cookie, updatedCookie);
   Object.assign(cookie.fides_meta, extraDetails); // save extra details to meta (i.e. consentMethod)
 
-  // 2. Update the window.Fides object
+  // 2. Dispatch a "FidesUpdating" event with the new cookie
+  dispatchFidesEvent("FidesUpdating", cookie, options.debug, extraDetails);
```

Comment:
There are some other options where this could be. The `updateCookie` is an async function so it could potentially take a while, but it doesn't and I thought it made sense to send the `FidesUpdating` event with the new cookie.